nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -205,16 +202,6 @@ def setup(self, trainer: pl.Trainer) -> None:
             assert self.model is not None
             self.model = self._layer_sync.apply(self.model)
 
-        datamodule = getattr(trainer, "datamodule", None)
-        if not self.data_sampler and hasattr(datamodule, "data_sampler"):
-            self.data_sampler = datamodule.data_sampler
-            self.data_sampler.setup(self.cluster_environment.global_rank())
-            datamodule.reconfigure_limit_batches()
```

Comment:
Is there a reason why this line was removed?

Reply:
I pulled them into the megatron setup callback. Just pushed a new commit to add the last line (I think that one is new)